Superset installation build fails using Docker on MS-Windows - fatal errors result in non-working system

A clear and concise description of what the bug is.

### Expected results

Clone and load the SuperSet demonstration Docker container by Run the following commands on the Superset Home page:

```git clone https://github.com/apache/incubator-superset/
cd incubator-superset
# you can run this command everytime you need to start superset now:
docker-compose up```

"After several minutes for superset initialization to finish, you can open a browser and view http://localhost:8088 to start your journey."


### Actual results

The build gets to Step 11/32 and throws an error due to an incompatible version of a JavaScript package.  Updating caniuse-lite is a red herring and does not help.  Running `npm update` does not help either.  Allocating more RAM as suggested by others does not help either.  

```
Step 11/32 : RUN cd /app/superset/assets         && npm run build         && rm -rf node_modules
 ---> Running in 4d072668b171

> superset@0.999.0-dev build /app/superset/assets
> cross-env NODE_OPTIONS=--max_old_space_size=8192 NODE_ENV=production webpack --mode=production --colors --progress

clean-webpack-plugin: /app/superset/assets/dist has been removed.
Starting type checking service...
Using 1 worker with 2048MB memory limit
 92% chunk asset optimization OptimizeCssAssetsWebpackPluginBrowserslist: caniuse-lite is outdated. Please run next command `npm update`
 92% chunk asset optimization TerserPluginnpm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! superset@0.999.0-dev build: `cross-env NODE_OPTIONS=--max_old_space_size=8192 NODE_ENV=production webpack --mode=production --colors --progress`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the superset@0.999.0-dev build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-12-06T01_40_49_400Z-debug.log
ERROR: Service 'superset' failed to build: The command '/bin/sh -c cd /app/superset/assets         && npm run build         && rm -rf node_modules' returned a non-zero code: 1
```
#### How to reproduce the bug

1. Go to 'https://superset.incubator.apache.org/installation.html#superset-installation-and-initialization'
2. Execute Docker container gitclone instruction: git clone https://github.com/apache/incubator-superset/
3. change directory to superset ```cd incubator-superset```

# you can run this command everytime you need to start superset now:
```docker-compose up```

4. wait about 15-20 minutes for Docker container to build executable file from Python, Javascript, Postgres and other components. 
4. See error in terminal

### Environment

Docker Desktop Community Edition v2.1.0.5 (40693) on Windows 10 

- superset version: `superset version`  Latest as of this posting
- python version: `python --version`   3.7.1
- node.js version: `node -v`  10.16.0
- npm version: `npm -v` 6.9.0

### Checklist

- [ x ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ x]  I have reproduced the issue with at least the latest released version of superset.
- [ x ] I have checked the issue tracker for the same issue and I haven't found one similar.

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -29,6 +29,7 @@ x-superset-volumes: &superset-volumes
   - ./docker/pythonpath_dev:/app/pythonpath
   - ./superset:/app/superset
   - ./superset-frontend:/app/superset-frontend
+  - node_modules:/app/superset-frontend/node_modules
   - superset_home:/app/superset_home
 
 version: "3.7"
@@ -84,6 +85,8 @@ services:
 
 volumes:
   superset_home:
+    external: false
+  node_modules:
     external: false
   db_home:
     external: false

